File: python/get_user.py
import json
import boto3

dynamodb = boto3.resource('dynamodb')
table = dynamodb.Table('Users')

def lambda_handler(event, context):
    user_id = event['pathParameters']['userId']
    response = table.get_item(Key={'userId': user_id})

    if 'Item' not in response:
        return {"statusCode": 404, "body": json.dumps({"message": "User not found"})}

    return {
        "statusCode": 200,
        "body": json.dumps(response['Item'])
    }

File: node/getUser.js
const db = require('./utils/dbClient');

exports.handler = async (event) => {
  const { userId } = event.pathParameters;

  const result = await db.get({
    TableName: 'Users',
    Key: { userId }
  }).promise();

  return {
    statusCode: result.Item ? 200 : 404,
    body: JSON.stringify(result.Item || { message: 'User not found' }),
  };
};
